pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/BookieRackAffinityMapping.java

```diff
@@ -80,29 +80,49 @@ public static MetadataStore createMetadataStore(Configuration conf) throws Metad
             store = (MetadataStore) storeProperty;
         } else {
             String url;
-            String metadataServiceUri = ConfigurationStringUtil.castToString(conf.getProperty("metadataServiceUri"));
-            if (StringUtils.isNotBlank(metadataServiceUri)) {
+            // First get from the Pulsar broker side
+            String metadataStoreUrl = ConfigurationStringUtil.castToString(conf.getProperty("metadataStoreUrl"));
```

Comment:
This logic is used for bk auto-recovery, and it doesn't make sense to get it from the Pulsar broker side. If we use it with this logic, we need to configure `metadataStoreUrl` in the `conf/bookkeeper.conf`

Reply:
The current BookieRackAffinityMapping policy is implemented on the pulsar side and its configuration is configured to the pulsar's zk (metadatastore). 
It would be best to migrate the Metadata BookieRackAffinityMapping strategy to the bookkeeper project so that bookkeeper-only businesses can have this capability as well.